.github/workflows/CI-pr.yml

```diff
@@ -0,0 +1,135 @@
+name: CI-pr
```

Comment:
You can still keep the workflow name as `CI`, but split it across two different files to run differently for different triggers.